src/Microsoft.Graph.Cli.Core/Authentication/ClientCertificateCredentialFactory.cs

```diff
@@ -74,19 +76,26 @@ internal static bool TryGetCertificateFromStore(string certificateNameOrThumbPri
         try
         {
             store.Open(OpenFlags.ReadOnly);
-
-            // If using a certificate with a trusted root you do not need to FindByTimeValid, instead:
-            // currentCerts.Find(X509FindType.FindBySubjectDistinguishedName, certName, true);
-            X509Certificate2Collection signingCerts = store.Certificates.Find(X509FindType.FindByTimeValid, DateTime.Now, false)
-                .Find(isThumbPrint ? X509FindType.FindByThumbprint : X509FindType.FindBySubjectDistinguishedName, certificateNameOrThumbPrint, false);
+            X509Certificate2Collection signingCerts = store.Certificates
+                .Find(X509FindType.FindByTimeValid, DateTime.Now, true)
+                .Find(isThumbPrint ? X509FindType.FindByThumbprint : X509FindType.FindBySubjectDistinguishedName,
+                    certificateNameOrThumbPrint, true);
             if (signingCerts.Count == 0)
             {
                 result = false;
             }
             else
             {
                 // Return the first certificate in the collection, has the right name and is current.
-                certificate = signingCerts.OrderByDescending(static c => c.NotBefore).FirstOrDefault();
+                foreach (var cert in signingCerts)
+                {
+                    // Use this certificate if it became valid after the currently selected one.
+                    if (certificate is null || cert.NotBefore > certificate.NotBefore)
```

Comment:
we should probably add a condition here to make sure the not before is lower than the current date, in case future certificates have been provisioned.

Reply:
According to the documentation, it considers everything including the trust chain https://learn.microsoft.com/en-us/dotnet/api/system.security.cryptography.x509certificates.x509certificate2collection.find?view=net-8.0

Reply:
Btw, this `Find(X509FindType.FindByTimeValid, DateTime.Now, false)` should handle validity time, no?

Reply:
ah correct, I hadn't thought of that in my other comment.
Those APIs feel ancient....